src/colorize.cr

```diff
@@ -226,36 +226,52 @@ module Colorize
       io << blue
     end
   end
+
+  # A text decoration.
+  #
+  # Note that not all text decorations are supported in all terminals.
+  # When a text decoration is not supported, it will leave the text unaffected.
+  @[Flags]
+  enum Mode
+    # Makes the text bold.
+    Bold = 1
+    # Makes the text color bright.
+    Bright = 1
+    # Dims the text color.
+    Dim
+    # Underlines the text.
+    Underline
+    # Makes the text blink slowly.
+    Blink
+    # Swaps the foreground and background colors of the text.
+    Reverse
+    # Makes the text invisible.
+    Hidden
+
+    def code
+      case self
```

Comment:
Couldn't this be turned into an exhaustive enum?

Reply:
Exhaustive case doesn't work for flag enums.

Reply:
I think it just highlights that it's a strange thing to do - picking out results for individual values in a flags enum. It has a tendency of misunderstanding - for an actual combined flags value the other sub-values will be ignored.
Also consider not making this method public.
I address this in https://github.com/oprypin/crystal/commit/5956803b29c408a64f9ac3c00810151b57e2d047